Leanplum-SDK/Classes/Internal/Leanplum.m

```diff
@@ -1548,8 +1547,8 @@ + (void)defineAction:(NSString *)name ofKind:(LeanplumActionKind)kind withArgume
     }
 
     LP_TRY
+    // This should be done elsewhere. I don't know where.
```

Comment:
I think it's fine to initialize it here if needed. However, I don't think we need the `BOOL initializedMessageTemplates` to exist at all. We can just call `[LPMessageTemplatesClass sharedTemplates]` to initialize. If it is already initialized, it'll be a no-op

Reply:
The issue is that we _cannot_ initialize it here if it is initializing, because we get the deadlock. How do we know to skip that? The BOOL. If we move this initialization out of the chain for its initialization, then we don't need the BOOL anymore. I don't know where that is though.